reflectapi/src/codegen/python.rs

```diff
@@ -0,0 +1,4117 @@
+    }
+
+    // Check recursively in type arguments
+    type_ref
+        .arguments
+        .iter()
+        .any(|arg| type_reference_references_type(arg, target_type))
+}
+
+/// Recursively check if a type uses reflectapi::Option
+fn type_uses_reflectapi_option(type_def: &reflectapi_schema::Type) -> bool {
+    match type_def {
+        reflectapi_schema::Type::Struct(s) => {
+            // Check all fields in the struct
+            for field in s.fields.iter() {
+                if type_reference_uses_reflectapi_option(&field.type_ref) {
+                    return true;
+                }
+            }
+        }
+        reflectapi_schema::Type::Enum(_) => {
+            // Enums don't contain reflectapi::Option
+        }
+        reflectapi_schema::Type::Primitive(_) => {
+            // Primitives don't contain reflectapi::Option
+        }
+    }
+    false
+}
+
+/// Check if a type reference uses reflectapi::Option
+fn type_reference_uses_reflectapi_option(type_ref: &TypeReference) -> bool {
+    // Check if this is directly a reflectapi::Option
+    if type_ref.name == "reflectapi::Option" {
+        return true;
+    }
+
+    // Recursively check type arguments
+    type_ref
+        .arguments
+        .iter()
+        .any(type_reference_uses_reflectapi_option)
+}
+
+fn build_implemented_types() -> HashMap<String, String> {
+    let mut types = HashMap::new();
+
+    // Primitive types
+    types.insert("i8".to_string(), "int".to_string());
+    types.insert("i16".to_string(), "int".to_string());
+    types.insert("i32".to_string(), "int".to_string());
+    types.insert("i64".to_string(), "int".to_string());
+    types.insert("u8".to_string(), "int".to_string());
+    types.insert("u16".to_string(), "int".to_string());
+    types.insert("u32".to_string(), "int".to_string());
+    types.insert("u64".to_string(), "int".to_string());
+    types.insert("f32".to_string(), "float".to_string());
+    types.insert("f64".to_string(), "float".to_string());
+    types.insert("bool".to_string(), "bool".to_string());
+    types.insert("String".to_string(), "str".to_string());
+    types.insert("std::string::String".to_string(), "str".to_string());
+
+    // Collections with full Rust type names (using modern lowercase hints for Python 3.9+)
+    types.insert("std::vec::Vec".to_string(), "list[T]".to_string());
+    types.insert(
+        "std::collections::HashMap".to_string(),
+        "dict[K, V]".to_string(),
+    );
+    types.insert(
+        "std::collections::BTreeMap".to_string(),
+        "dict[K, V]".to_string(),
+    );
+    types.insert("std::option::Option".to_string(), "T | None".to_string());
+    types.insert("std::result::Result".to_string(), "T | E".to_string());
+
+    // ReflectAPI specific types
+    types.insert(
+        "reflectapi::Option".to_string(),
+        "ReflectapiOption[T]".to_string(),
+    );
+    types.insert(
+        "reflectapi::Empty".to_string(),
+        "ReflectapiEmpty".to_string(),
```

Comment:
I'm surprised you needed a custom type here - wouldn't this just map to `None`

Reply:
I think it is semantically correct. `None` might be inappropriate because:
- Empty structs serialize to `{}` in JSON, not `null`
- HTTP APIs distinguish between "no content" and JSON responses that are "empty object", and `null`.